DevEdu.API/DevEdu.Business/Services/TaskService.cs

```diff
@@ -9,39 +13,59 @@ public class TaskService : ITaskService
         private readonly ITaskRepository _taskRepository;
         private readonly ICourseRepository _courseRepository;
         private readonly IStudentAnswerOnTaskRepository _studentAnswerOnTaskRepository;
+        private readonly ITaskValidationHelper _taskValidationHelper;
 
-        public TaskService(ITaskRepository taskRepository, ICourseRepository courseRepository, IStudentAnswerOnTaskRepository studentAnswerOnTaskRepository)
+        public TaskService(
+            ITaskRepository taskRepository,
+            ICourseRepository courseRepository,
+            IStudentAnswerOnTaskRepository studentAnswerOnTaskRepository, 
+            ITaskValidationHelper taskValidationHelper
+        )
         {
             _taskRepository = taskRepository;
             _courseRepository = courseRepository;
             _studentAnswerOnTaskRepository = studentAnswerOnTaskRepository;
+            _taskValidationHelper = taskValidationHelper;
         }
 
-        public TaskDto GetTaskById(int id) => _taskRepository.GetTaskById(id);
-
-        public TaskDto GetTaskWithCoursesById(int id)
+        public TaskDto GetTaskById(int id)
         {
+            // check if task exists
+            _taskValidationHelper.CheckTaskExistence(id);
             var taskDto = _taskRepository.GetTaskById(id);
-            taskDto.Courses = _courseRepository.GetCoursesToTaskByTaskId(id);
             return taskDto;
         }
 
-        public TaskDto GetTaskWithAnswersById(int id)
+        public TaskDto GetTaskByIdWithValidation(int id, int userId)
```

Comment:
rename this method to GetTaskById (as you are using the method from 31st row only here)

Reply:
Not only here, that's why i made those two methods separately